File: BOJ/1000~9999/6603_로또/6603_로또.java
import java.io.*;
import java.util.*;

public class Main {
  public static BufferedReader br;
  public static BufferedWriter bw;
  public static int n;
  public static int size = 6;
  public static int[] arr;
  public static int[] box;
  public static void main(String argc[]) throws IOException{
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));
    
    while(true) {
      StringTokenizer st = new StringTokenizer(br.readLine());
      n = Integer.parseInt(st.nextToken());
      if(n == 0) {
    	  break;
      }
      arr = new int[n];
      box = new int[6];
      for(int i=0; i<n; ++i){
        arr[i] = Integer.parseInt(st.nextToken());
      }
      for(int i=0; i<n; ++i){
        DFS(i, 0);
      }
      bw.write("\n");
    }
    bw.close();
  }
  public static void DFS(int index, int cnt) throws IOException{
    box[cnt] = arr[index];
    if(cnt+1 == size){
      for(int i=0; i<size; ++i){
        bw.write(box[i] + " ");
      }
      bw.write("\n");
      return;
    }
    for(int i = index+1; i<n; ++i){
      DFS(i, cnt+1);
    }
  }
}

File: BOJ/1000~9999/6603_로또/6603_로또.cpp
#include <iostream>
#include <algorithm>
#include <cstring>
using namespace std;

int n;
int arr[20];
int com[20];

void DFS(int next, int depth)
{
  if (depth == 6)
  {
    for (int i = 0; i < 6; ++i)
    {
      cout << com[i] << ' ';
    }
    cout << '\n';
    return;
  }

  for (int i = next; i < n; ++i)
  {
    com[depth] = arr[i];
    DFS(i + 1, depth + 1);
  }
}

int main(int argc, char *argv[])
{
  while (1)
  {
    cin >> n;

    if (n == 0)
    {
      break;
    }

    for (int i = 0; i < n; ++i)
    {
      cin >> arr[i];
    }

    DFS(0, 0);
    cout << '\n';
  }

  return 0;
}